### ISSUE TITLE
safe.directory '*' is not accepted

### ISSUE THREAD

**Issue by laktak** [libgit2/libgit2#6391]
I'm a user of gitui and the owner says the following bug should be reported here (see https://github.com/extrawurst/gitui/issues/1293). I hope it's OK that the repro steps are done using gitui.

**Describe the bug**

git works on mounted directories if you use
```
git config --global --add safe.directory '*'
```

But gitui will throw an error:
```
invalid path
please run gitui inside of a non-bare git repository
```

I can work around the issue by using 
```
git config --global --add safe.directory /mnt/...
```
but I have to do this for every directory.

**To Reproduce**
- run `git config --global --add safe.directory '*'`
- go to a mounted directory
- run `git clone https://github.com/extrawurst/gitui`
- cd into gitui and run `git status` # this will work
- run `gitui` # this will give an error

**Expected behavior**
gitui should use the same options as git (if this is a problem of a dependency please let me know so I can report it there)

**Context (please complete the following information):**
 - arch
 - gitui 0.20.1 (from arch repos)

**Comment by ethomson** [comment]
Thanks for the report. `safe.directory` is going to shape up like `core.autocrlf`; a gift that keeps on giving. 😬 

Apparently wildcard support was introduced in git 2.35.1, which we can emulate easily enough.

**Comment by ethomson** [comment]
Fixed by #6429
